mesonbuild/build.py

```diff
@@ -1113,9 +1119,49 @@ def __init__(self, name, subdir, subproject, is_cross, sources, objects, environ
             self.filename += '.' + self.suffix
         self.outputs = [self.filename]
 
+        # The import library this target will generate
+        self.import_filename = None
+        # The import library that Visual Studio would generate (and accept)
+        self.vs_import_filename = None
+        # The import library that GCC would generate (and prefer)
+        self.gcc_import_filename = None
+
+        # if implib:true appears, this target is linkwith:-able, but that only
+        # means something on Windows platforms.
+        self.is_linkwithable = False
+        if 'implib' in kwargs and kwargs['implib']:
+            self.is_linkwithable = True
+            if for_windows(is_cross, environment) or for_cygwin(is_cross, environment):
+                self.vs_import_filename = '{0}.lib'.format(self.name)
+                self.gcc_import_filename = 'lib{0}.exe.a'.format(self.name)
```

Comment:
autotools doesn't know anything about these libraries, so you end up writing the -out-implib flag in the Makefile.am yourself.  I think this is the conventional way to name these files, but it could vary, so I guess control is needed of the implib name.

No, -lfoo won't work, you'd need to write -lfoo.exe (unlike when linking with a DLL where -lfoo could be silently converted to foo.dll.a).  But it's unclear anyone would actually want that, as there's no ambiguity if a static or shared library is wanted.  This is a different category of thing.